sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1774,14 +1774,17 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
           val zoneId = getZoneId(SQLConf.get.sessionLocalTimeZone)
           toLiteral(stringToTimestamp(_, zoneId), TimestampType)
         case "INTERVAL" =>
-          val interval = try {
+          val intervalWithoutSign = try {
             IntervalUtils.fromString(value)
           } catch {
             case e: IllegalArgumentException =>
               val ex = new ParseException("Cannot parse the INTERVAL value: " + value, ctx)
               ex.setStackTrace(e.getStackTrace)
               throw ex
           }
+          val interval = if (ctx.sign != null && ctx.sign.getText == "-") {
+            IntervalUtils.negate(intervalWithoutSign)
+          } else intervalWithoutSign
           Literal(interval, CalendarIntervalType)
```

Comment:
We can use one line here and revert 1777 by reusing the generalized `applySign`.
```scala
-          val interval = if (ctx.sign != null && ctx.sign.getText == "-") {
-            IntervalUtils.negate(intervalWithoutSign)
-          } else intervalWithoutSign
-          Literal(interval, CalendarIntervalType)
+          Literal(applySign(ctx.sign, interval), CalendarIntervalType)
```